tests/providers/google/cloud/hooks/test_cloud_run.py

```diff
@@ -0,0 +1,78 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import unittest
+from unittest import mock
+from uuid import UUID
+
+from airflow.providers.google.cloud.hooks.cloud_run import CloudRunJobHook
+
+
+TASK_ID = "test-cloud-run-operator"
+JOB_NAME = "test-cloud-run-job"
+CLOUD_RUN_JOBS_STRING = "airflow.providers.google.cloud.hooks.cloud_run.{}"
+TEST_PROJECT = "test-project"
+MOCK_UUID = UUID("cf4a56d2-8101-4217-b027-2af6216feb48")
+MOCK_UUID_PREFIX = str(MOCK_UUID)[:8]
+EXECUTION_ID = f"{JOB_NAME}-{MOCK_UUID_PREFIX}"
+
+
+class TestCloudRunJobsHook(unittest.TestCase):
+    def setUp(self):
+        self.cloud_run_job_hook = CloudRunJobHook(gcp_conn_id="google_cloud_default")
```

Comment:
@VinceLegendre we on the way of migration all tests to native `pytest`. Main target right now remove all subclasses of  `unittest.TestCase`.

I would be very appreciate if you do not use `unittest.TestCase`. For setup/teardown test you could use fixtures (preferable way) or [xunit-style set-up style](https://docs.pytest.org/en/7.1.x/how-to/xunit_setup.html)

Reply:
Hi @Taragolis , thanks for the review :+1: !
I'll work on the migration from unittest to pytest in the coming days

Reply:
Fill free to ping me in Slack if you need any assistance with migration from `unittests` to `pytest` or you could use some previous PRs as sample: https://github.com/apache/airflow/pulls?q=is%3Apr+pytest+is%3Aclosed

Reply:
Hi @Taragolis , just delivered a new version without the use of `unittest.TestCase`.
 I may mention that I used `pytest-subtests` module to replace unitttests subtests.

Reply:
Soooory for late response. 

`pytest-subtests` have additional issues like it mark succeeded main test even if subtest failed.
I will add in separate comment how to do the same thing without additional dependency, and also in most cases subtests are useless

Reply:
Tests updated :heavy_check_mark:

Reply:
FYI Completion requirements were discussed in this issue : https://github.com/apache/airflow/issues/24730.

Maybe we could consider all the Jobs CRUD operations in a later PR, when the official JobsClient can be installed properly?

Opened to suggestions here :wink: